src/melee.cpp

```diff
@@ -2273,6 +2251,39 @@ std::vector<special_attack> player::mutation_attacks(Creature &t) const
 
 std::string melee_message( const ma_technique &tec, player &p, const dealt_damage_instance &ddi )
 {
+    // Those could be extracted to a json
+
+    // Three last values are for low damage
+    static const std::array<std::string, 6> player_stab = {{
+        "You impale %s", "You gouge %s", "You run %s through",
```

Comment:
This might clarify, _() actually does two things, one is it "captures" the string, which happens in a step where gettext processes (not compiles!) all the source code, all this does is build a list of literal strings enclosed in _().
The second thing kicks in at runtime, where _() becomes a macro that passes the string to the gettext library for translation, this may happen at the same site as where the capture occurs, or it can also happen elsewhere.
As long as the _() invocation happens after the gettext library is initialized (in practice, within any function as I outlined), it will be able to do the translation.